snomed/com.b2international.snowowl.snomed.fhir/src/com/b2international/snowowl/snomed/fhir/SnomedConceptMapApiProvider.java

```diff
@@ -299,8 +288,8 @@ public TranslateResult translate(ComponentURI componentURI, TranslateRequest tra
 			.all()
 			.filterByActive(true)
 			.filterByTypes(CONCEPT_MAP_TYPES)
-			.filterByMapTargetComponentType(terminologyComponentIdAsInt)
-			.filterByReferencedComponentType(SnomedTerminologyComponentConstants.CONCEPT)
+			.filterByMapTargetComponentType(targetSystem)
```

Comment:
Is this going to work if `targetSystem` is not modified by the `if` statements above?

Reply:
Yes, that's the plan. I've fixed the incorrect map target in this case and also the search request method infinite loop.